### PR TITLE
Training-27: Hadoop Training Slides

### PR DESCRIPTION
**Pull request by greatgautam**
Create Apache Hadoop Slides for adding to Apache Training

### CODE REVIEW

**Comment by RyanSkraba** [comment]
Hello!  I think you accidently merged the proposed TRAINING-28 (Hive) slides into the TRAINING-27 (Hadoop) PR.

I don't know if we have best practices yet -- but I find it very useful to have the JIRA link in the PR name (and optionally in the commit messages).

Can you remove the `.DS_Store` files from the PR?

File: content/ApacheHadoop/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+(input) <k1, v1> -> map -> <k2, v2> -> combine -> <k2, v2> -> reduce -> <k3, v3> (output)
+
+== HDFS Architecture
+HDFS has a master/slave architecture. An HDFS cluster consists of:
+- single NameNode, a master server that manages the file system namespace and regulates access to files by clients
+- one of more of DataNodes used for storage and serving read and write requests from the file system’s clients
+
+== HDFS New Features
+New features and improvements are regularly implemented in HDFS. The following is a subset of useful features in HDFS:
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/ApacheHadoop/src/main/asciidoc/index.adoc]: I have updated it.

File: content/ApacheHadoop/src/main/asciidoc/index.adoc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+The MapReduce framework operates exclusively on <key, value> pairs, that is, the framework views the input to the job as a set of <key, value> pairs and produces a set of <key, value> pairs as the output of the job, conceivably of different types.
+
+The key and value classes have to be serializable by the framework and hence need to implement the Writable interface. Additionally, the key classes have to implement the WritableComparable interface to facilitate sorting by the framework.
+
+Input and Output types of a MapReduce job:
+
+(input) <k1, v1> -> map -> <k2, v2> -> combine -> <k2, v2> -> reduce -> <k3, v3> (output)
+
+== HDFS Architecture
+HDFS has a master/slave architecture. An HDFS cluster consists of:
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/ApacheHadoop/src/main/asciidoc/index.adoc]: Added new line.

File: content/ApacheHadoop/src/main/asciidoc/index.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Two major layers
+
+- Processing layer (MapReduce)
+- Storage layer (Hadoop Distributed File System)
+
+== MapReduce
+
+Hadoop MapReduce is a software framework for easily writing applications which process vast amounts of data (multi-terabyte data-sets) in-parallel on large clusters (thousands of nodes) of commodity hardware in a reliable, fault-tolerant manner. 
+
+The MapReduce framework consists of:
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/ApacheHadoop/src/main/asciidoc/index.adoc]: Added a newline. Please confirm if it looks good.

**Comment by greatgautam** [comment]
Hi Ryan,
Thanks. I have added commands as well. Please review.

regards,
Gautam

**Comment by RyanSkraba** [comment]
It builds correctly, and the slides look correctly formatted!  I agree this could serve as a starting point for someone to build on -- as I mentioned, Hadoop is a vast subject!

Calling any committer to take a look :D

Have you been following this https://github.com/apache/incubator-training/pull/43 ?

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  limitations under the License.
+
+////
+:revealjs_progress: true
+:revealjs_slidenumber: true
+:sourcedir: ../java
+
+== What is Apache Hadoop?
+
+Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc]: Updated.

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+Two main layers:
+
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc:28]: I have added the new text.

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+- Processing layer (MapReduce)
+- Storage layer (Hadoop Distributed File System)
+
+== MapReduce
+
+Hadoop MapReduce is a software framework for easily writing applications which process vast amounts of data (multi-terabyte data-sets) in-parallel on large clusters (thousands of nodes) of commodity hardware in a reliable, fault-tolerant manner. 
+
+The MapReduce framework consists of:
+
+- single master ResourceManager
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc]: Updated!

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+- one worker NodeManager per cluster-node
+- MRAppMaster per application 
+
+== Hadoop Distributed File System
+
+Hadoop Distributed File System (HDFS) is a distributed file system designed to run on commodity hardware. 
+
+- similar to other distributed file systems
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc]: Thanks, I took care of it/

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+== What is Apache Hadoop?
+
+Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
+
+Two main layers:
+
+- Processing layer (MapReduce)
+- Storage layer (Hadoop Distributed File System)
+
+== MapReduce
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc:36]: Updated

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+- Safemode: an administrative mode for maintenance.
+- fsck: a utility to diagnose health of the file system, to find missing files or blocks.
+- fetchdt: a utility to fetch DelegationToken and store it in a file on the local system.
+- Balancer: tool to balance the cluster when the data is unevenly distributed among DataNodes.
+- Upgrade and rollback: after a software upgrade, it is possible to rollback to HDFS’ state before the upgrade in case of unexpected problems.
+
+
+== HDFS Special Nodes
+
+Main nodes in HDFS are as follows:
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc:93]: Thanks, I took care of it.

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+Input and Output types of a MapReduce job:
+
+(input) <k1, v1> -> map -> <k2, v2> ->
+combine -> <k2, v2> ->
+reduce -> <k3, v3> (output)
+
+== HDFS Architecture
+
+HDFS has a master/slave architecture. An HDFS cluster consists of:
+
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc:74]: Added the diagram.

File: content/Hadoop/src/main/asciidoc/index.adoc [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+== HDFS Commands
+
+All HDFS commands are invoked by the bin/hdfs script.
+Running the hdfs script without any arguments prints the description for all commands.
+
+Usage: hdfs [SHELL_OPTIONS] COMMAND [GENERIC_OPTIONS] [COMMAND_OPTIONS]
+
+
+== Common HDFS Commands
```
[the remark this answers is not in the document]
Comment (greatgautam) [reply content/Hadoop/src/main/asciidoc/index.adoc:127]: Yes, added for YARN.

**Comment by greatgautam** [comment]
Hi,
I have updated the PR for Apache Hadoop slides. Please review and merge if
it looks good.

https://github.com/apache/incubator-training/pull/63

thanks,
Gautam

**Comment by RyanSkraba** [comment]
Thanks for the contribution!  This looks encouraging.

**Comment by greatgautam** [comment]
Hi,
I am new to the JIRA flow. Since the PR is merged, when can we close the
following issue in JIRA:

https://issues.apache.org/jira/browse/TRAINING-27


thanks,
Gautam

**Pull request merged**